src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1609,6 +1646,88 @@ RebalanceTableShards(RebalanceOptions *options, Oid shardReplicationModeOid)
 }
 
 
+/*
+ * RebalanceTableShardsBackground rebalances the shards for the relations
+ * inside the relationIdList across the different workers. It does so using our
+ * background job+task infrastructure.
+ */
+static void
+RebalanceTableShardsBackground(RebalanceOptions *options, Oid shardReplicationModeOid)
+{
+	char transferMode = LookupShardTransferMode(shardReplicationModeOid);
+	EnsureReferenceTablesExistOnAllNodesExtended(transferMode);
+
+	if (list_length(options->relationIdList) == 0)
+	{
+		return;
+	}
+
+	Oid relationId = InvalidOid;
+	char *operationName = "rebalance";
+	if (options->drainOnly)
+	{
+		operationName = "move";
+	}
+
+	foreach_oid(relationId, options->relationIdList)
+	{
+		AcquireRebalanceColocationLock(relationId, operationName);
+	}
+
+	List *placementUpdateList = GetRebalanceSteps(options);
+
+	if (list_length(placementUpdateList) == 0)
+	{
+		return;
+	}
+
+	/* find the name of the shard transfer mode to interpolate in the scheduled command */
+	Datum shardTranferModeLabelDatum =
+		DirectFunctionCall1(enum_out, shardReplicationModeOid);
+	char *shardTranferModeLabel = DatumGetCString(shardTranferModeLabelDatum);
+
+	if (list_length(placementUpdateList) == 0)
+	{
+		return;
+	}
+
+	/* schedule planned moves */
+	int64 jobId = CreateBackgroundJob("rebalance", "Rebalance colocation group ...");
+
+	PlacementUpdateEvent *move = NULL;
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+	bool first = true;
+	int64 prevJobId = 0;
+	foreach_ptr(move, placementUpdateList)
+	{
+		resetStringInfo(&buf);
+
+		appendStringInfo(&buf,
+						 "SELECT citus_move_shard_placement(%ld,%s,%u,%s,%u,%s)",
+						 move->shardId,
+						 quote_literal_cstr(move->sourceNode->workerName),
+						 move->sourceNode->workerPort,
+						 quote_literal_cstr(move->targetNode->workerName),
+						 move->targetNode->workerPort,
+						 quote_literal_cstr(shardTranferModeLabel));
+
+		BackgroundTask *task = ScheduleBackgroundTask(jobId, GetUserId(), buf.data,
+													  first ? 0 : 1, &prevJobId);
+		prevJobId = task->taskid;
+		first = false;
+	}
+
+	/*
+	 * This uses the first relationId from the list, it's only used for display
+	 * purposes so it does not really matter which to show
+	 */
+
+	/* ExecutePlacementUpdates(placementUpdateList, shardReplicationModeOid, "Moving"); */
+	/* FinalizeCurrentProgressMonitor(); */
+}
```

Comment:
it would be good to give some output, at minimum the job ID and some notices about what the job ID means (which tables to check) and whether any moves were planned

Reply:
I think notices explaining what the job id means is somewhat noisy and an anti pattern. What would the benefit of notices be, compared to good documentation on our docs page?

Reply:
@mulander 's doc had notices with the job ID and a hint with what to do next, which seems like a useful thing for a function that's likely to be used in an interactive session

again, I think job ID should be minimum, but it also seems important to tell the user whether something was scheduled, or how many tasks were scheduled

Reply:
The main reason to do so is to make it clear to the user that the command is not finished yet, but is running in the background. Having some useful info on how to check its status would be good. With the current foreground rebalancer we've had users that didn't know about get_rebalance_progress. I think with the background rebalancer this will be even more likely to happen, if the user doesn't get any output. The design doc had the following output: 

```
NOTICE:  Scheduling...
NOTICE:  Scheduled as job 1337.
DETAIL:  Rebalance scheduled as background job 1337.
HINT:  To monitor progress, run: SELECT details FROM citus_rebalance_status();
```

I don't think the `Scheduling...` line is particularly helpful but the rest seems like useful info

Reply:
We had the same take on `Scheduling...` I have copied the other messages mostly.
As discussed in chat the `HINT` is slightly different to point users to `pg_dist_background_task` and `get_rebalance_progress`.